linera-core/src/client.rs

```diff
@@ -1803,3 +1975,70 @@ where
             .await
     }
 }
+
+/// The outcome of trying to commit a list of operations to the chain.
+#[derive(Debug)]
+pub enum ClientOutcome<T> {
+    /// The operations were committed successfully.
+    Committed(T),
+    /// We are not the round leader and cannot do anything. Try again at the specified time or
+    /// or whenever the round or block height changes.
+    WaitForTimeout(RoundTimeout),
+}
+
+#[derive(Debug)]
+pub struct RoundTimeout {
+    pub timestamp: Timestamp,
+    pub current_round: Round,
+    pub next_block_height: BlockHeight,
+}
+
+impl<T> ClientOutcome<T> {
+    pub fn unwrap(self) -> T {
+        match self {
+            ClientOutcome::Committed(t) => t,
+            ClientOutcome::WaitForTimeout(_) => panic!(),
+        }
+    }
+
+    pub fn expect(self, msg: &'static str) -> T {
+        match self {
+            ClientOutcome::Committed(t) => t,
+            ClientOutcome::WaitForTimeout(_) => panic!("{}", msg),
+        }
+    }
+
+    pub fn map<F, S>(self, f: F) -> ClientOutcome<S>
+    where
+        F: FnOnce(T) -> S,
+    {
+        match self {
+            ClientOutcome::Committed(t) => ClientOutcome::Committed(f(t)),
+            ClientOutcome::WaitForTimeout(timeout) => ClientOutcome::WaitForTimeout(timeout),
+        }
+    }
+
+    #[allow(clippy::result_large_err)]
+    fn try_map<F, S>(self, f: F) -> Result<ClientOutcome<S>, ChainClientError>
+    where
+        F: FnOnce(T) -> Result<S, ChainClientError>,
+    {
+        match self {
+            ClientOutcome::Committed(t) => Ok(ClientOutcome::Committed(f(t)?)),
+            ClientOutcome::WaitForTimeout(timeout) => Ok(ClientOutcome::WaitForTimeout(timeout)),
+        }
+    }
+}
+
+/// The outcome of trying to commit a list of incoming messages and operations to the chain.
+#[derive(Debug)]
+enum ExecuteBlockOutcome {
+    /// A block with the messages and operations was committed.
+    Executed(Certificate),
+    /// A different block was already proposed and got committed. Check whether the messages and
+    /// operations are still suitable, and try again at the next block height.
+    OtherBlock(Certificate),
```

Comment:
maybe `Conflict` or `ConflictingBlock`?

Reply:
I renamed it to `Conflict` and fixed the test by boxing large fields of `ChainManagerInfo`.